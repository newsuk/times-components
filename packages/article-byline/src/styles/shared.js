import styleguide from "@times-components/styleguide";

const { colours, fontFactory } = styleguide();
const shared = {
  link: {
    color: colours.functional.action,
    ...fontFactory({
      font: "supporting",
      fontSize: "cardMeta"
    })
  },
  text: {
    ...fontFactory({
      font: "supporting",
      fontSize: "cardMeta"
    }),
    color: colours.functional.secondary,
    flexDirection: "row"
  }
};

export default shared;
