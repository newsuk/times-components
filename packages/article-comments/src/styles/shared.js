import styleguide from "@times-components/styleguide";

const { colours, fontFactory, spacing } = styleguide();
const styles = {
  button: {
    marginBottom: spacing(0),
    marginTop: spacing(5),
    maxWidth: 215
  },
  container: {
    alignItems: "center",
    borderStyle: "solid",
    borderTopColor: colours.functional.keyline,
    borderTopWidth: 1,
    marginBottom: spacing(5),
    width: "100%"
  },
  headline: {
    color: colours.functional.primary,
    ...fontFactory({
      font: "headline",
      fontSize: "commentsHeadline"
    }),
    maxWidth: 315,
    paddingBottom: spacing(2),
    paddingTop: spacing(6),
    textAlign: "center"
  },
  link: {
    color: colours.functional.action
  },
  supporting: {
    ...fontFactory({
      font: "supporting",
      fontSize: "tertiary"
    }),
    color: colours.functional.secondary,
    maxWidth: 280,
    textAlign: "center"
  }
};

export default styles;
