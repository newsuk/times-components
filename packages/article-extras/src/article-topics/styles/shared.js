import { colours, tabletWidth } from "@times-components/styleguide";

const sharedStyles = {
  topicsContainer: {
    borderTopColor: colours.functional.keyline,
    borderTopWidth: 1
  },
  topicsContainerTablet: {
    alignSelf: "center",
    width: tabletWidth
  },
  topicsMetaContainer: {
    justifyContent: "flex-start"
  }
};

export default sharedStyles;
