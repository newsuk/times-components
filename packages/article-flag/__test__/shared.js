/* eslint-env jest */

import "react-native";
import React from "react";
// eslint-disable-next-line import/no-extraneous-dependencies
import renderer from "react-test-renderer";

import ArticleFlag, {
  NewArticleFlag,
  UpdatedArticleFlag,
  ExclusiveArticleFlag,
  SponsoredArticleFlag
} from "../article-flag";

module.exports = () => {
  it("renders correctly", () => {
    const tree = renderer.create(<ArticleFlag title="articleFlag" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders New flag correctly", () => {
    const tree = renderer.create(<NewArticleFlag />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders Updated flag correctly", () => {
    const tree = renderer.create(<UpdatedArticleFlag />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders Exclusive flag correctly", () => {
    const tree = renderer.create(<ExclusiveArticleFlag />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders Sponsored flag correctly", () => {
    const tree = renderer.create(<SponsoredArticleFlag />).toJSON();
    expect(tree).toMatchSnapshot();
  });
};
