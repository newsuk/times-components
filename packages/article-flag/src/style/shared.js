import styleguide from "@times-components/styleguide";

const { fontFactory } = styleguide();
const styles = {
  diamond: {
    marginRight: 5
  },
  title: {
    ...fontFactory({
      font: "bodyRegularSmallCaps",
      fontSize: "cardMetaMobile"
    }),
    fontWeight: "400",
    letterSpacing: 1.4
  },
  view: {
    alignItems: "center",
    flexDirection: "row"
  }
};

export default styles;
