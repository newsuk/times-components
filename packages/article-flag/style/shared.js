import { fonts } from "@times-components/styleguide";

const styles = {
  view: {
    flexDirection: "row",
    alignItems: "center"
  },
  diamond: {
    marginRight: 5,
    marginBottom: 1
  },
  title: {
    fontFamily: fonts.bodyRegularSmallCaps,
    fontSize: 12,
    fontWeight: "400",
    letterSpacing: 1.4
  }
};

export default styles;
