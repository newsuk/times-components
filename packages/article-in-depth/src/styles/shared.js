import styleguide from "@times-components/styleguide";

const { colours, fontFactory, spacing } = styleguide();
const sharedStyles = {
  articleHeadline: {
    ...fontFactory({
      font: "headline",
      fontSize: "headline"
    }),
    color: colours.functional.brandColour,
    marginBottom: spacing(2),
    textAlign: "center"
  },
  container: {
    alignItems: "center",
    paddingBottom: spacing(9),
    paddingHorizontal: spacing(4),
    paddingTop: spacing(12)
  },
  datePublication: {
    ...fontFactory({
      font: "supporting",
      fontSize: "cardMeta"
    }),
    color: colours.functional.secondary,
    marginTop: spacing(1)
  },
  flag: {
    flexDirection: "row",
    marginBottom: spacing(3),
    marginTop: spacing(1)
  },
  flagContainer: {
    marginRight: spacing(3)
  },
  label: {
    marginBottom: spacing(2)
  },
  leadAssetContainer: {
    marginBottom: spacing(4)
  },
  meta: {
    flexDirection: "row",
    flexWrap: "wrap"
  },
  metaContainer: {
    alignItems: "center",
    borderBottomWidth: 1,
    borderColor: colours.functional.keyline,
    borderTopWidth: 1,
    marginBottom: spacing(5),
    marginHorizontal: spacing(2),
    paddingBottom: 6,
    paddingTop: spacing(2)
  },
  standFirst: {
    ...fontFactory({
      font: "headlineRegular",
      fontSize: "smallestHeadline"
    }),
    color: colours.functional.primary,
    marginBottom: spacing(3),
    paddingHorizontal: spacing(2),
    textAlign: "center"
  }
};

export default sharedStyles;
