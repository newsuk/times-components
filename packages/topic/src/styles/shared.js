import styleguideFactory from "@times-components/styleguide";

const { colours, fontFactory, spacing } = styleguideFactory();
const styles = {
  wrapper: {
    width: "100%",
    alignItems: "center",
    flexDirection: "column",
    paddingTop: spacing(8),
    paddingHorizontal: spacing(2),
    backgroundColor: colours.functional.backgroundPrimary,
    borderBottomColor: colours.functional.border,
    borderBottomWidth: 1,
    minHeight: 90
  },
  container: {
    alignItems: "center"
  },
  name: {
    ...fontFactory({ font: "headline", fontSize: "pageHeadline" }),
    color: colours.functional.brandColour,
    textAlign: "center"
  },
  description: {
    ...fontFactory({ font: "body", fontSize: "tertiary" }),
    textAlign: "center",
    color: colours.functional.primary
  },
  divider: {
    borderStyle: "solid",
    borderTopColor: colours.functional.keyline,
    borderTopWidth: 1,
    marginBottom: spacing(5),
    width: 200,
    justifyContent: "center"
  }
};

export default styles;
