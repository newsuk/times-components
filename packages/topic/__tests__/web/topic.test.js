import shared from "../shared";

describe("Topic tests on web: ", () => {
  shared();
});
