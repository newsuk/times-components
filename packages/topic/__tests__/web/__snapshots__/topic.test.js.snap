// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Topic tests on web should render HeadContainer correctly 1`] = `
<div
  className="responsive-styles__component-fbPFZZ CHBUp"
/>
`;

exports[`Topic tests on web should render ResponsiveDivider correctly 1`] = `
<div
  className="responsive-styles__component-fbPFZZ sAowN"
/>
`;

exports[`Topic tests on web should render ResponsiveName correctly 1`] = `
<div
  className="responsive-styles__component-fbPFZZ kfmhxW"
  dir="auto"
/>
`;

exports[`Topic tests on web should render correctly 1`] = `
<div>
  <div
    className="responsive-styles__component-fbPFZZ CHBUp"
  >
    <div>
      <h1
        aria-label="topic-name"
        className="responsive-styles__component-fbPFZZ kfmhxW"
        data-testid="topic-name"
        dir="auto"
        role="heading"
      >
        Animals
      </h1>
    </div>
  </div>
</div>
`;

exports[`Topic tests on web should render correctly when loading 1`] = `<div />`;

exports[`Topic tests on web should render correctly with a description 1`] = `
<div>
  <div
    className="responsive-styles__component-fbPFZZ CHBUp"
  >
    <div>
      <h1
        aria-label="topic-name"
        className="responsive-styles__component-fbPFZZ kfmhxW"
        data-testid="topic-name"
        dir="auto"
        role="heading"
      >
        Animals
      </h1>
      <div
        className="responsive-styles__component-fbPFZZ sAowN"
      />
      <div
        data-testid="topic-description"
        dir="auto"
      >
        Animals are multicellular eukaryotic organisms.
      </div>
    </div>
  </div>
</div>
`;
